### DISCUSSION TITLE
Asset preview

### DISCUSSION THREAD

**Discussion by AhmedAbdelwahed01** [atteneder/glTFast#401] [Ideas]
If there any utility to return Texture2D as asset preview for the gltf file @atteneder

**Comment by atteneder** [comment]
Hi,

Can you elaborate:

- In what context?
- You mean a preview rendering of the glTF?

**Reply by AhmedAbdelwahed01** [reply]
Yes exactly like you got a preview image when you import 3d model fbx in the assets you got like preview image on it

**Reply by AhmedAbdelwahed01** [reply]
![image](https://user-images.githubusercontent.com/29849559/169047145-bb470cf8-0cc2-4df8-a3ab-46b062aaf9a6.png)

**Reply by AhmedAbdelwahed01** [reply]
this preview for example has Engine utility to make it like that after importing

**Reply by atteneder** [reply]
Here's a side-by-side of an FBX and a glTF. Looks like this is already working:
<img width="251" alt="image" src="https://user-images.githubusercontent.com/2593014/169068515-a2079007-4006-4a7a-86cc-5c94931691d5.png">
Sometimes the preview rendering takes a second or two though.

**Reply by AhmedAbdelwahed01** [reply]
Sorry for misunderstanding i mean if there any Runtime API that returns the asset preview  like this one    "AssetPreview.GetAssetPreview()"
but this one form unity for editor utility

**Reply by atteneder** [reply]
There is none and I think this shouldn't be a feature of glTFast.

You can, however, generate this yourself by rending to a render texture, right? Have a look at `GltfBoundsAsset` to see how you can get the assets' bounds in order to place a camera properly.

**Reply by AhmedAbdelwahed01** [reply]
Thanks will try it
